src/toolbars/CutCopyPasteToolBar.cpp

```diff
@@ -0,0 +1,174 @@
+
+enum {
+   TBCutID,
+   TBCopyID,
+   TBPasteID,
+   TBNumButtons
+};
+
+constexpr int first_TB_ID = 21300;
+
+static const ButtonList CutCopyPasteToolbarButtonList = {
+   { TBCutID,   wxT("Cut"),   XO("Cut")   },
+   { TBCopyID,  wxT("Copy"),  XO("Copy")  },
+   { TBPasteID, wxT("Paste"), XO("Paste") }
+};
+
+IMPLEMENT_CLASS(CutCopyPasteToolBar, ToolBar);
+
+////////////////////////////////////////////////////////////
+/// Methods for CutCopyPasteToolBar
+////////////////////////////////////////////////////////////
+
+BEGIN_EVENT_TABLE( CutCopyPasteToolBar, ToolBar )
+   EVT_COMMAND_RANGE(TBCutID+first_TB_ID,
+                      TBCutID+first_TB_ID + TBNumButtons - 1,
+                      wxEVT_COMMAND_BUTTON_CLICKED,
+                      CutCopyPasteToolBar::OnButton )
+END_EVENT_TABLE()
+
+//Standard constructor
+CutCopyPasteToolBar::CutCopyPasteToolBar( AudacityProject &project )
+: ToolBar(project, CutCopyPasteBarID, XO("Cut/Copy/Paste"), wxT("Cut/Copy/Paste"))
+, mButtons{ this, project, CutCopyPasteToolbarButtonList, TBNumButtons, first_TB_ID }
+{
+}
+
+CutCopyPasteToolBar::~CutCopyPasteToolBar()
+{
+}
+
+void CutCopyPasteToolBar::Create(wxWindow * parent)
+{
+   ToolBar::Create(parent);
+   UpdatePrefs();
+}
+
+void CutCopyPasteToolBar::AddButton(
+   teBmps eEnabledUp, teBmps eEnabledDown, teBmps eDisabled,
+   int id, const TranslatableString &label, bool toggle)
+{
+   auto r = mButtons.CreateButton(eEnabledUp, eEnabledDown, eDisabled, id, label, toggle);
+   mToolSizer->Add(r);
+}
+
+void CutCopyPasteToolBar::Populate()
+{
+   SetBackgroundColour( theTheme.Colour( clrMedium  ) );
+   MakeButtonBackgroundsSmall();
+
+   Add(mToolSizer = safenew wxGridSizer(2, 2, 1, 1));
+
+   /* Buttons */
+   // Tooltips match menu entries.
+   // We previously had longer tooltips which were not more clear.
+   AddButton(bmpCut, bmpCut, bmpCutDisabled, TBCutID,
+      XO("Cut"));
+   AddButton(bmpCopy, bmpCopy, bmpCopyDisabled, TBCopyID,
+      XO("Copy"));
+   AddButton(bmpPaste, bmpPaste, bmpPasteDisabled, TBPasteID,
+      XO("Paste"));
+
+   mButtons.SetEnabled(TBPasteID, false);
+
+   RegenerateTooltips();
+}
+
+void CutCopyPasteToolBar::UpdatePrefs()
+{
+   RegenerateTooltips();
+
+   // Set label to pull in language change
+   SetLabel(XO("Cut/Copy/Paste"));
+
+   // Give base class a chance
+   ToolBar::UpdatePrefs();
+}
+
+void CutCopyPasteToolBar::RegenerateTooltips()
+{
+   mButtons.RegenerateTooltips();
+}
+
+void CutCopyPasteToolBar::EnableDisableButtons()
+{
+   mButtons.EnableDisableButtons();
+}
+
+void CutCopyPasteToolBar::OnButton(wxCommandEvent & event)
+{
+   mButtons.OnButton(event);
+}
+
+static RegisteredToolbarFactory factory{ CutCopyPasteBarID,
+   []( AudacityProject &project ){
+      return ToolBar::Holder{ safenew CutCopyPasteToolBar{ project } }; }
+};
+
+#include "ToolManager.h"
+
+namespace {
+AttachedToolBarMenuItem sAttachment{
+   /* i18n-hint: Clicking this menu item shows the toolbar for editing */
```

Comment:
I suggest an OrderingHint to control the position of insertion of this menu item.  Did design specify it?  Maybe it should come after the edit toolbar, not at the bottom of the menu.